From 481b4ac7fd1b06521eaf6984109221a16d05d664 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Fri, 6 Oct 2023 10:57:13 -0400
Subject: [PATCH 36/53] Catch NPE in SystemUI

No idea why it triggers, but it crashes SystemUI on boot for my TV
device
---
 .../com/android/systemui/keyguard/KeyguardViewMediator.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
index 0de61a2652be..1643c1535477 100644
--- a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
+++ b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
@@ -3482,7 +3482,11 @@ public class KeyguardViewMediator implements CoreStartable, Dumpable,
         Trace.beginSection("KeyguardViewMediator#handleMotifyStartedWakingUp");
         synchronized (KeyguardViewMediator.this) {
             if (DEBUG) Log.d(TAG, "handleNotifyWakingUp");
-            mKeyguardViewControllerLazy.get().onStartedWakingUp();
+            try {
+                mKeyguardViewControllerLazy.get().onStartedWakingUp();
+            } catch(Throwable t) {
+                android.util.Log.e("PHH", "handleNotifyStartedWakingUp crashed bip", t);
+            }
         }
         Trace.endSection();
     }
-- 
2.34.1

